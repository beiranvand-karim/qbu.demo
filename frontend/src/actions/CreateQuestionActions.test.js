/* global expect, describe, it, beforeEach */
import configureStore from 'redux-mock-store'
import {
   createQuestionBegin,
   createQuestionSuccess,
   createQuestionError
} from './CreateQuestionActions'


describe('CreateQuestionActions', () => {

   const mockStore = configureStore();
   const store = mockStore();

   beforeEach(() => {
      store.clearActions();
   });

   it('should matches the create question begin snapshot', () => {
      store.dispatch(createQuestionBegin());
      expect(store.getActions()).toMatchSnapshot();
   });

   it('should matches the create question success snapshot', () => {
      const data = 'some data';
      store.dispatch(createQuestionSuccess(data));
      expect(store.getActions()).toMatchSnapshot();
   });

   it('should matches the create question error snapshot', () => {
      const error = 'some error';
      store.dispatch(createQuestionError(error));
      expect(store.getActions()).toMatchSnapshot();
   });
});
