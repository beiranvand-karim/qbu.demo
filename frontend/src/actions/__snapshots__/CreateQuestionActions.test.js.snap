// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateQuestionActions should matches the create question begin snapshot 1`] = `
Array [
  Object {
    "type": "CREATE_QUESTION_BEGIN",
  },
]
`;

exports[`CreateQuestionActions should matches the create question error snapshot 1`] = `
Array [
  Object {
    "payload": "some error",
    "type": "CREATE_QUESTION_ERROR",
  },
]
`;

exports[`CreateQuestionActions should matches the create question success snapshot 1`] = `
Array [
  Object {
    "payload": "some data",
    "type": "CREATE_QUESTION_SUCCESS",
  },
]
`;
